targeted/dns-email-spoofing.js

```diff
@@ -0,0 +1,188 @@
+ * @param {Object.<HttpMessage>} msg    - The HttpMessage Object being scanned
+ * @param {String} policy               - The policy name
+ * @param {RegExp} policyRegex          - The regex expression for detecting the policy
+ * @param {String} url                  - The URL against which script has been invoked
+ */
+function testForPolicy(msg, policy, policyRegex, url){
+    var newReq = msg.cloneRequest();
+	// Fetch TXT DNS records for root domain
+	var fetchedTxtRecords = fetchRecords(newReq, policy);
+	logger("Checking for presence of " + policy + " records.");
+	checkIfPolicy(fetchedTxtRecords, policyRegex, policy, url, newReq);
+}
+
+/**
+ * Checks if a specific DNS TXT record policy exists or not,
+ * Raises an alert if the policy is not present
+ * 
+ * @param {Object.object} txtRecords	- The fetched DNS Records response
+ * @param {RegExp} policyRegex 			- The regex expression for detecting the policy
+ * @param {String} policyName 			- The policy name
+ * @param {String} metaData 			- The extra meta data that will be sent in otherInfo field
+ * @param {Object.<HttpMessage>} msg 	- The HttpMessage Object being scanned
+ */
+function checkIfPolicy(txtRecords, policyRegex, policyName, url, msg){
+
+	var cweId = 290;
+	var wascId = 12;
+    // All TXT records are under the "Answer" key in the response object
+	var foundPolicy = checkForPolicy(txtRecords["Answer"], policyRegex);
+
+	if(foundPolicy !== true){
+		var alertName = policyName + " Records not configured";
+		var alertSol = policyName + " Records should be configured properly on domain to prevent email spoofing. "
+		var alertDesc = "Phishing and email spam are the biggest opportunities for hackers to enter the network. "
+		+	"If a single user clicks on some malicious email attachment, it can compromise an entire enterprise with ransomware, "
+		+	"cryptojacking scripts, data leakages or privilege escalation exploits. This can be prevented with the help of "
+		+	policyName + " records."
+        customAlert(
+            pluginid,
+            2, // risk:       0: info, 1: low, 2: medium, 3: high
+            3, // confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+            alertName,
+            alertDesc,
+            alertSol,
+            cweId,
+            wascId,
+            msg,
+            url
+        )
+	}
+}
+
+/**
+ * Function to fetch DNS TXT records over DoH
+ * @param {Object.<HttpMessage>} msg - The HttpMessage Object being scanned
+ * @param {String} policy 			 - The policy name to fetch records for.
+ * @return {Object.object}			 - The fetched records.
+ */
+function fetchRecords(msg, policy) {
+	var domain = msg.getRequestHeader().getURI().getHost();
+	if(domain.startsWith("www")){
+		domain = domain.replace("www.","")
+	}
+	if (policy === "DMARC") {
+		domain = "_dmarc." + domain;
+	}
+    var scheme = "https://";
+    var path = "/resolve?name=" + domain + "&type=TXT";
+    var targetUri = scheme + providerAddress + path;
```

Comment:
Scheme is only used once, why not just use the literal directly?
```suggestion
    var path = "/resolve?name=" + domain + "&type=TXT";
    var targetUri = "https://" + providerAddress + path;
```